lib/hades_web/views/user_view.ex

```diff
@@ -10,4 +10,12 @@ defmodule HadesWeb.UserView do
   def render("user.json", %{user: user}) do
     %{id: user.id, email: user.email, name: user.name, is_admin: user.is_admin}
   end
+
+  def render("user_status.json", %{user: user}) do
```

Comment:
Is it possible to use the `when` clause here for the `user.is_active` checking? Then we avoid the `if/else`, WDYT?

Reply:
Yes we can. There a lot of options beyond the `if/else`. Could you give me an example of your suggestion?

Reply:
I am not sure if this is a valid statement in Elixir, but I am thinking in something like:

```
def render("user_status.json", %{user: user}) when user.is_active, do: %{message: "User is active"}
def render("user_status.json", %{user: user}), do: %{message: "User is inactive"}
```

Reply:
I got this error when trying the example above: `cannot invoke remote function user.is_active/0 inside guard`

Reply:
Is it maybe possible to decompose the map ?
```elixir
def render("user_status.json", %{user: %{is_active: true}}) do: %{message: "User is active"}
```
Not sure this is possible.
In any case, I think @rafaels88 suggestion is a nice improvement.

Reply:
I also tried grabbing just the status (`render("user_status.json", status: user.is_active)`) and checking if the status is true using the same approach with guards but I got this:
```
== Compilation error in file lib/hades_web/views/user_view.ex ==
** (SyntaxError) lib/hades_web/views/user_view.ex:18: unexpected keyword "do:". In case you wanted to write a "do" expression, you must either separate the keyword argument with comma or use do-blocks. For example, the following construct:

    if some_condition? do
      :this
    else
      :that
    end

is syntactic sugar for the Elixir construct:

    if(some_condition?, do: :this, else: :that)

where "some_condition?" is the first argument and the second argument is a keyword list.

Syntax error before: 'do:'
    (elixir) lib/kernel/parallel_compiler.ex:121: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/1

```

Reply:
@rafaels88 What do you think about using `case`?
i.e.:
```
case status do
  true ->
    # active
  false
    # inactive
end
```

Reply:
But if we really don't want a block inside the view, maybe we can use something like this:
```
def render("user_status.json", %{user: user}) do
    %{message: "User is #{if user.is_active, do: "active", else: "inactive"}"}
  end
```